prefix-dev/pixi
# feat: add matchspec caching

## Overview

This PR adds ability to load already existing tool from the disk.

So if previously we installed `boltons==1.0.0`, and now we are requesting `boltons *`, we will preload it.
if we request `boltons >= 1.1.0`, we will install a new one. 
More details are inside the `Matches` trait : https://github.com/conda/rattler/blob/main/crates/rattler_conda_types/src/match_spec/mod.rs#L364

it also add a new clean command for cache, `pixi clean cache --tool` to clean build tools envs.


How to test it:
* remove `rattler/cache/cached-build-tool-envs-v0` if exist.
* run `pixi install -vvv` for rich_example. You will see in tracing::debug `not found any existing environment` message, and also `cached-build-tool-envs-v0` should now have a folder for installed tool.
* run again `pixi install -vvv` . You will see now `reusing existing environment in` and no new folders should be in `cached-build-tool-envs-v0`.
* change build-backend in pixi for this `build-backend = { name = "pixi-build-python", version = "<0.1.0dev20241203154923" }`. This will make pixi to search for older pixi, that don't satisfisy already installed one.
run `pixi install -vvv` and now you will see `not found any existing environment for` and a new folder for this new tool should be in `cached-build-tool-envs-v0`.
